### PR TITLE
KAA-1131 Programming guide page

### PR DESCRIPTION
**Pull request by forGGe**

### CODE REVIEW

File: doc/Programming-guide/index.md [hunk 1]
```diff
@@ -2,40 +2,23 @@
...
-
----
-
-Copyright (c) 2014-2015, [CyberVision, Inc.](http://www.cybervisiontech.com/)
-
-Document generated by Confluence on Dec 22, 2015 16:12
-
-[Atlassian](http://www.atlassian.com/)
\ No newline at end of file
+Finally, if anything goes wrong you can always refer to [the troubleshooting section]({{root_url}}Programming-guide/Debugging-and-troubleshooting/) or ask for help in our [community forum](https://groups.google.com/group/kaaproject)
```
Comment (Slash32) [inline doc/Programming-guide/index.md]: 'dot' is missing

File: doc/Programming-guide/index.md [hunk 2]
```diff
@@ -2,40 +2,23 @@
...
+{% assign root_url = page.url | split: '/'%}
+{% capture root_url  %} /{{root_url[1]}}/{{root_url[2]}}/{% endcapture %}
 
-The intention of this guide is to introduce Kaa platform programming basics to you. You will have to roll up your sleeves with Kaa (at least so we hope), so come prepared. In order for the guide to be useful, make sure that you have a development environment set up, including a Kaa server instance. We recommend that you use [Kaa Sandbox](Sandbox) with this guide, but you can [use a Kaa cluster](Installation-guide) instead, if you like. We also recommend skimming through the [Getting started guide](Getting-started) beforehand.
+The intention of this guide is to introduce Kaa platform programming basics to you.
+You will have to roll up your sleeves with Kaa (at least so we hope), so come prepared.
+In order for the guide to be useful, make sure that you have a development environment set up, including a Kaa server instance.
+We recommend that you use Kaa Sandbox, which is covered in [the Getting Started guide]({{root_url}}/Getting-started/), but you can [use a Kaa cluster]({{root_url}}/Administration-guide/System-installation/) instead, if you like.
```
Comment (balldir) [inline doc/Programming-guide/index.md]: > [use a Kaa cluster]({{root_url}}/Administration-guide/System-installation/)
> Should it be a link to "Cluster setup guide" ?

File: doc/Programming-guide/index.md [hunk 3]
```diff
@@ -2,40 +2,23 @@
...
 
-For each of the Kaa functional areas, programming guide provides the following information:
+# Sections overview
 
-* Introduction and a brief function overview
-* References to the corresponding [design reference sections](Design-reference)
-* Description of the required configuration and references to corresponding topics in the [Admin UI guide](Administration-UI-guide)
-* References to relevant REST API methods in the [REST API documentation](Admin-REST-API)
-* Endpoint SDK API description and examples of use for different platforms
+It is best to dive directly into the practice [by creating basic Kaa application]({{root_url}}Getting-started/Your-first-Kaa-application/). You will learn a lot about Kaa during this basic training.
```
Comment (balldir) [inline doc/Programming-guide/index.md]: > [by creating basic Kaa application]({{root_url}}Getting-started/Your-first-Kaa-application/)

Broken link

File: doc/Programming-guide/index.md [hunk 4]
```diff
@@ -2,40 +2,23 @@
...
+With fresh new knowlegle you could start familiarize with [a system overview of the Kaa platform]({{root_url}}Programming-guide/System-overview/) and then go through [application-level features that our platform provides to you]({{root_url}}Programming-guide/Key-platform-features/)
 
-# Integrating Kaa into your project
+You then can advance your skills and create industrial-level projects by using [client-side SDK](Programming-guide/Using-Kaa-endpoint-SDKs/) provided in different languages and for different platforms.
```
Comment (balldir) [inline doc/Programming-guide/index.md]: > [client-side SDK](Programming-guide/Using-Kaa-endpoint-SDKs/)

broken link

File: doc/Programming-guide/index.md [hunk 5]
```diff
@@ -2,40 +2,23 @@
...
+Make sure you do not miss our [REST API documentation](Programming-guide/Server-REST-APIs/) which provides overview of a great automation tool.
```
Comment (balldir) [inline doc/Programming-guide/index.md]: > [REST API documentation](Programming-guide/Server-REST-APIs/)

Broken link

File: doc/Programming-guide/index.md (hunk 1 above)
Comment (forGGe) [inline doc/Programming-guide/index.md]: Fixed

File: doc/Programming-guide/index.md (hunk 2 above)
Comment (forGGe) [inline doc/Programming-guide/index.md]: Done

File: doc/Programming-guide/index.md (hunk 3 above)
Comment (forGGe) [inline doc/Programming-guide/index.md]: Fixed

File: doc/Programming-guide/index.md (hunk 4 above)
Comment (forGGe) [inline doc/Programming-guide/index.md]: Fixed

File: doc/Programming-guide/index.md (hunk 5 above)
Comment (forGGe) [inline doc/Programming-guide/index.md]: Fixed

**Comment by forGGe** [comment]
@balldir , @Slash32 please re-review

**Comment by Slash32** [comment]
Looks good to me

File: doc/Programming-guide/index.md [hunk 6]
```diff
@@ -2,40 +2,24 @@
...
+# Sections overview
 
-* Introduction and a brief function overview
-* References to the corresponding [design reference sections](Design-reference)
-* Description of the required configuration and references to corresponding topics in the [Admin UI guide](Administration-UI-guide)
-* References to relevant REST API methods in the [REST API documentation](Admin-REST-API)
-* Endpoint SDK API description and examples of use for different platforms
+It is best to dive directly into the practice [by creating basic Kaa application]({{root_url}}Programming-guide/Your-first-Kaa-application/).
+You will learn a lot about Kaa during this basic training.
+With fresh new knowlegle you could start familiarize with [a system overview of the Kaa platform]({{root_url}}Programming-guide/System-overview/) and then go through [application-level features that our platform provides to you]({{root_url}}Programming-guide/Key-platform-features/)
```
Comment (Polinaa) [inline doc/Programming-guide/index.md:20]: knowledge

File: doc/Programming-guide/index.md [hunk 7]
```diff
@@ -2,40 +2,24 @@
...
 
-# Integrating Kaa into your project
+You then can advance your skills and create industrial-level projects by using [client-side SDK]({{root_url}}Programming-guide/Using-Kaa-endpoint-SDKs/) provided in different languages and for different platforms.
+Make sure you do not miss our [REST API documentation]({{root_url}}Programming-guide/Server-REST-APIs/) which provides overview of a great automation tool.
```
Comment (Polinaa) [inline doc/Programming-guide/index.md:23]: an overview

File: doc/Programming-guide/index.md (hunk 6 above)
Comment (Polinaa) [inline doc/Programming-guide/index.md:20]: 'dot' is missing

File: doc/Programming-guide/index.md [hunk 8]
```diff
@@ -2,40 +2,24 @@
...
+{% capture root_url  %} /{{root_url[1]}}/{{root_url[2]}}/{% endcapture %}
 
-The intention of this guide is to introduce Kaa platform programming basics to you. You will have to roll up your sleeves with Kaa (at least so we hope), so come prepared. In order for the guide to be useful, make sure that you have a development environment set up, including a Kaa server instance. We recommend that you use [Kaa Sandbox](Sandbox) with this guide, but you can [use a Kaa cluster](Installation-guide) instead, if you like. We also recommend skimming through the [Getting started guide](Getting-started) beforehand.
+The intention of this guide is to introduce Kaa platform programming basics to you.
+You will have to roll up your sleeves with Kaa (at least so we hope), so come prepared.
+In order for the guide to be useful, make sure that you have a development environment set up, including a Kaa server instance.
+We recommend that you use Kaa Sandbox, which is covered in [the Getting Started guide]({{root_url}}Getting-started/), but you can [use a Kaa cluster]({{root_url}}Administration-guide/System-installation/Cluster-setup/) instead, if you like.
```
Comment (Polinaa) [inline doc/Programming-guide/index.md:14]: Should `Single node installation` be mentioned here too?

File: doc/Programming-guide/index.md (hunk 7 above)
Comment (forGGe) [inline doc/Programming-guide/index.md:23]: +

File: doc/Programming-guide/index.md (hunk 6 above)
Comment (forGGe) [inline doc/Programming-guide/index.md:20]: +

File: doc/Programming-guide/index.md (hunk 8 above)
Comment (forGGe) [inline doc/Programming-guide/index.md:14]: +

**Comment by forGGe** [comment]
@polinaa comments addressed in #868

**Pull request merged**